src/main/java/fopbot/KarelWorld.java

```diff
@@ -320,21 +320,27 @@ protected boolean isCoinInField(int x, int y) {
     }
 
     /**
-     * Returns {@code true} if this world is visible on the graphical user interface.
+     * Returns {@code true} if this world is visible on the graphical user interface. Returns
+     * {@code false} if this world is running in headless mode.
      *
-     * @return {@code true} if this world is visible on the graphical user interface.
+     * @return {@code true} if this world is visible on the graphical user interface. Returns
+     *     {@code false} if this world is running in headless mode
      */
     public boolean isVisible() {
-        return guiFrame != null && guiFrame.isVisible();
+        return !GraphicsEnvironment.isHeadless() && guiFrame != null && guiFrame.isVisible();
     }
 
     /**
      * Sets the visibility of the world on the graphical user interface to the specified visibility
-     * value.
+     * value. Does nothing if the world is running in headless mode.
      *
      * @param visible if {@code true} this world will be visible on the graphical user interface
      */
     public void setVisible(boolean visible) {
+        if (GraphicsEnvironment.isHeadless()) {
+            System.err.println("Cannot set world visible in headless mode. In future versions this will throw an exception.");
```

Comment:
This should either already throw an exception or just silently fail. It doesn't make sense to add the exception in future versions because right now (before the winter semester) is the chance to make breaking changes.

I would just say just keep this failing silently, there's no reason to throw an exception here

Reply:
fixed by https://github.com/tudalgo/fopbot/pull/32/commits/3a1a07b7fd7b511297319a18be02265157a08b45